crates/polars-io/src/csv/write/write_impl/serializer.rs

```diff
@@ -121,13 +123,23 @@ fn integer_serializer<I: NativeType + itoa::Integer>(array: &PrimitiveArray<I>)
     })
 }
 
-fn float_serializer_no_precision<I: NativeType + ryu::Float>(
+fn float_serializer_no_precision<I: NativeType + ryu::Float + LowerExp + NumCast>(
```

Comment:
You should create a new serializer. The idea of the separate serializers is to have no branches and thus are faster.

Reply:
I wondered about that, but I wasn’t able to see a performance difference in my application (I guess because the branch condition isn’t changing so the CPU predictor is quite good). Nevertheless, eliminating branches would be better. Do you prefer copying and pasting the functions with small changes, or is it OK to use generic parameters to resolve the branches at compile time?

Reply:
If it can be done cleanly with a generic that is preferred, but if the copied code is small and the generic solution isn't really nice, go for the copy. ;)

Reply:
Ok, thanks! My attempt with generics only saved a few lines of code and made the whole thing a bit less readable, so using separate functions seems better.